app/src/main/java/com/lagradost/cloudstream3/ui/download/DownloadFragment.kt

```diff
@@ -280,4 +295,84 @@ class DownloadFragment : Fragment() {
 
         fixPaddingStatusbar(binding?.downloadRoot)
     }
+}
+
+class SwipeToDeleteCallback(private val adapter: DownloadHeaderAdapter) :
+    ItemTouchHelper.SimpleCallback(ItemTouchHelper.LEFT, 0) {
+
+    private var isActive = false
+
+    override fun onMove(
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        target: RecyclerView.ViewHolder
+    ): Boolean {
+        return false
+    }
+
+    override fun onSwiped(viewHolder: RecyclerView.ViewHolder, direction: Int) {
+        val position = viewHolder.bindingAdapterPosition
+        val item = adapter.cardList[position]
+
+        CoroutineScope(Dispatchers.Main).launch {
+            item.child?.let { clickEvent ->
+                handleDownloadClick(
+                    DownloadClickEvent(
+                        DOWNLOAD_ACTION_DELETE_FILE,
+                        clickEvent
+                    )
+                ) {
+                    if (it) adapter.notifyItemRemoved(viewHolder.absoluteAdapterPosition)
+                    isActive = false
+                }
+            }
+        }
+    }
+
+    override fun onChildDraw(
+        c: Canvas,
+        recyclerView: RecyclerView,
+        viewHolder: RecyclerView.ViewHolder,
+        dX: Float,
+        dY: Float,
+        actionState: Int,
+        isCurrentlyActive: Boolean
+    ) {
+        super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive)
+
+        val deleteIcon: Drawable? = ContextCompat.getDrawable(recyclerView.context, R.drawable.ic_baseline_delete_outline_24)
+        val background: ColorDrawable = ColorDrawable(Color.RED)
+
+        val itemView = viewHolder.itemView
+        val backgroundCornerOffset = 20
+
+        val iconMargin = (itemView.height - deleteIcon?.intrinsicHeight!!) / 2
```

Comment:
Now that I think of this this is probably a bad idea and should check for null instead of always assuming not just because of Android sometimes. I will fix this later as well.

Reply:
And... fixed